universal-application-tool-0.0.1/app/views/LoginForm.java

```diff
@@ -12,13 +12,18 @@
   public Content render(Http.Request request) {
     return htmlContent(
         body(
+            h1("Error: You are not logged in")
+                .withCondHidden(!request.queryString("message").orElse("").equals("login")),
             h1("Log In"),
             form(
                     makeCsrfTokenInputTag(request),
                     textField("uname", "username", "Username"),
                     passwordField("pwd", "password", "Password"),
                     submitButton("login", "Submit"))
                 .withMethod("POST")
-                .withAction("/callback?client_name=FormClient")));
+                .withAction("/callback?client_name=FormClient"),
```

Comment:
suggestion (non-blocking): I found [these docs](https://www.playframework.com/documentation/2.8.x/JavaRouting#Call-to-action-generator-method) on query strings. It could be convenient to use a `client_name` param in the `callback` controller method. There's also a way to specify a default value.

Likewise for potentially adding a `message` param on the `loginForm` controller method. (Unless you need the entire `request` in these cases.)